### ISSUE TITLE
Will it be difficult to do the worker API?

### ISSUE THREAD

**Issue by mik-laj** [PolideaInternal/airflow#532]

**Comment by potiuk** [comment]
Do we need Worker API if we use messaging queues? Unless we consider messaging queue as an API (which we should)

**Comment by nuclearpinguin** [comment]
What do we mean by "Worker API"? Internal Worker-Scheduler/Airflow API or interface meant for external usage?

**Comment by mik-laj** [comment]
@nuclearpinguin  Internal API for communication between Airflow worker and all other components including database.  Dawid dreamed about it at the last meeting in Google ..

@potiuk Probably yes, because there must be a way to indicate that the task is started by worker. (set state to running). However, it is also worth checking other cases to see if they can be done. 

This is not a task related to a specific action point, but rather a research that will allow us to better discuss with our client. I want to better understand the problem in order to be able to present it to the client.

**Comment by nuclearpinguin** [comment]
> @potiuk Probably yes, because there must be a way to mark that the task is started by worker. (set state to running). However, it is also worth checking other cases to see if they can be done.
> 

I am not sure if this is necessary. Take a look at the AirflowExecutor I proposed, the task is set to running by executing task not by executor, so there is no need for Worker -> Executor communication.

**Comment by potiuk** [comment]
Yeah. I realise that. What I think about it - it's not Task->Executor communication, but it's more Task -> DB communication. It can be asynchronous and can be done via some Pub/Sub queue. There is no need to save state of the task to the DB as it is done now.

But I think we should have some entit(ies) that will subscribe to the messages sent by Tasks and update state of those tasks in Database. It should not be Executor. In fact this should be totally decoupled from the executor implementation - different Executors should have the same "Task state Subscriber" or whatever we name it.

**Comment by potiuk** [comment]
Also there is one interesting thing that I realised. If we change the Worker -> DB api to pub/sub based, it can allow All executors to run with Sqlite database. That could be great for testing really - because we could test all executors with sqlite DB
